### PR TITLE
Add basic Rails boot stats

### PR DESCRIPTION
**Pull request by jules2689**
Add Rails Boot tasks
This will:
1) Uninstall all rails and active support gems
2) Install the proper rails version
3) Create an app
4) Measure the time it takes to run specified scripts (usually a rake or rails task/command)

cc @jerryliu55 @kirs

### CODE REVIEW

File: rails_boot/benchmarks/driver.rb [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      scripts[version[0]].each do |script|
+        output = measure(script, version)
+        return unless output
+
+        request = Net::HTTP::Post.new('/benchmark_runs')
+        request.basic_auth(ENV["API_NAME"], ENV["API_PASSWORD"])
+
+        submit = {
+          'benchmark_type[category]' => output["label"],
+          'benchmark_type[script_url]' => "#{RAW_URL}#{version}",
```
Comment (rafaelfranca) [inline rails_boot/benchmarks/driver.rb:65]: Is this URL correct?

Comment (jules2689) [reply rails_boot/benchmarks/driver.rb:65]: As jerry pointed out, we dont have a script url actually, this should be removed.

Comment (rafaelfranca) [reply rails_boot/benchmarks/driver.rb:65]: Or point to this script. The idea of this is url is to people click in the link in the benchmark result and can see the script that was used to generate the result.

File: rails_boot/benchmarks/driver.rb [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+  def files
+    Pathname.glob("#{File.expand_path(File.dirname(__FILE__))}/bm_*")
+  end
+
+   def run_single(version)
+    system('gem uninstall rails --all --force')
+    system('gem uninstall activesupport --all --force')
+    system("gem install rails:#{version} --no-doc")
+    system("rails _#{version}_ new my_app_#{version}")
```
Comment (kirs) [inline rails_boot/benchmarks/driver.rb:43]: `rails new my_app_5_0 --skip-spring` and you won't need to use `DISABLE_SPRING` below

File: rails_boot/benchmarks/driver.rb [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+        request = Net::HTTP::Post.new('/benchmark_runs')
+        request.basic_auth(ENV["API_NAME"], ENV["API_PASSWORD"])
+
+        submit = {
+          'benchmark_type[category]' => output["label"],
+          'benchmark_type[script_url]' => "#{RAW_URL}#{version}",
+          'benchmark_type[digest]' => Digest::SHA2.hexdigest(version),
+          'benchmark_run[environment]' => "#{`ruby -v; gem -v`.strip}",
+          'repo' => 'bundler',
```
Comment (kirs) [inline rails_boot/benchmarks/driver.rb:68]: bundler?

**Comment by kirs** [comment]
Is there any reason why we're sending the PR to fork?
I think we should do the same as we do with rails/rails and send the patches upstream.
The benchmark to measure Rails boot would be quite valuable for rubybench.

**Comment by jules2689** [comment]
@kirs we want to see if this works before PRing upstream

**Pull request closed without merging**